New features for csv import plugin

aroralakshya opened the pull request:
Following new features have been added in the csv import plugin:
1. For a new import, user can now directly enter the table or database name at the time of import
2. Partial import. User can enter how many rows they want to import.
Signed-Off-By: Lakshay arora <arora.lakshya123@gmail.com>

Before submitting pull request, please check that every commit:

- [x] Has proper Signed-Off-By
- [x] Has commit message which describes it
- [x] Is needed on it's own, if you have just minor fixes to previous commits, you can squash them
- [x] Any new functionality is covered by tests

aroralakshya committed:
Following new features have been added in the csv import plugin:

1. For a new import, user can now directly enter the table or database name at the time of import
2. Partial import. User can enter how many rows they want to import.
3. Show the form to edit table structure after import(undecided).

Signed-Off-By: Lakshay arora <arora.lakshya123@gmail.com>

aroralakshya committed:
Make Dialog messages translatable, add partial import feature to tbl_import
Signed-Off-By: Lakshay arora <arora.lakshya123@gmail.com>

mauriciofauth commented on js/functions.js:
```diff
@@ -1005,6 +1005,29 @@ AJAX.registerOnload('functions.js', function () {
+            data: params,
+            success: function (data) {
+                if (data.success) {
+                    $("<h3>You can change the table structure here</h3>" + data.message).insertAfter($("#new_csv_db_name").parent());
```
I think you forgot to make this string translatable.

aroralakshya committed:
Add an option to provide table and database name when importing from csv file.
Signed-Off-By: Lakshay arora <arora.lakshya123@gmail.com>

aroralakshya committed:
New features for csv import plugin
Following new features have been added in the csv import plugin:
1. For a new import, user can now directly enter the table or database name at the time of import
2. Partial import. User can enter how many rows they want to import.
Signed-Off-By: Lakshay arora <arora.lakshya123@gmail.com>

nulll-pointer commented:
Applied changes and solved merge conflicts. I'll work on tests as soon as travis finishes.

aroralakshya committed:
Fix tests
Signed-Off-By: Lakshay arora <arora.lakshya123@gmail.com>

ibennetch commented:
I think you forgot to make "<h3>You can change the table structure here</h3>" translatable here, other than that I think you got all of the requested changes.

nulll-pointer commented:
Done

aroralakshya committed:
Remove options to view and change table structure after upload

Signed-off-by: Lakshay arora <arora.lakshya123@gmail.com>

mauriciofauth commented on js/import.js:
```diff
@@ -89,6 +91,20 @@ AJAX.registerOnload('import.js', function () {
+            }
+            if($("#text_csv_new_db_name").length > 0) {
+                var newDBName = $("#text_csv_new_db_name").val();
+                if(newDBName.length > 0 && $.trim(newDBName).length === 0) {
```
Some lines caused ESLint to fail. See: https://travis-ci.org/phpmyadmin/phpmyadmin/jobs/413992898#L742

nulll-pointer replied:
These linting errors are in many of my PRs, it says it can be fixed using --fix option, how can I use this, I even asked this on the mailing list but got no reply.

ibennetch replied:
Probably `./node_modules/.bin/eslint js/import.js --fix` will do what you're after. Sorry I missed your email!

mauriciofauth replied:
And if you do not have the `node_modules` directory, you need to run the `yarn install` command

nulll-pointer commented:
I've fixed lint and other tests, but I don't understand this new selenium error?

mauriciofauth commented:
Merged, thanks for your contribution!
